app/controllers/api/users.php

```diff
@@ -138,7 +138,9 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
                 $existingTarget = $dbForProject->findOne('targets', [
                     Query::equal('identifier', [$email]),
                 ]);
-                $user->setAttribute('targets', [...$user->getAttribute('targets', []), $existingTarget]);
+                if($existingTarget) {
+                    $user->setAttribute('targets', [...$user->getAttribute('targets', []), $existingTarget]);
```

Comment:
`setAttribute()` has support for appending. See the [function signature](https://github.com/utopia-php/database/blob/45def2f7c6bc5f631dbb67e5df0e8e7331af5f63/src/Database/Document.php#L215). and example:

https://github.com/appwrite/appwrite/blob/0376189dbcb791d6ea28fd69332d7df1b208f1c3/app/init.php#L1075-L1080

Reply:
Done